### PR TITLE
Issue 17541 fix container as file refs on copy site

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
The problem this PR intents to solve is basically that the copy site was replicating a lot of the stuff that existed on the original source-site. contentlets, templates, containers, pages, etc.. but the referenced elements on the page multi-tree were still the original elements from the source origin site.

The problem has been around for a long time but it became evident when containers as files were introduced. The copied references of the containers as files couldn't be displayed properly on the new site.

So this fix basically takes all the references that to the new copied elements and updates the Multi-Tree and the templates making them point to the new elements copied into the new site.

The new Containers as files were being treated by the copy functionality as regular containers and even though you would see all the respective files under /application/containers/ in the copy-site. They were getting copied there by the logic that replicates the file-assets. Meaning that the multi-tree on the new copy-page not only saw elements from the original site but the new copy-containers were getting created as regular containers. Not containers as files.

The copy order of the elements had to change a bit in order to fit the new Containers-as-files because by the time the containers are getting copied the folders didn't exist yet. So when an instance of a CAF was found we needed to make sure the proper folder structure was already in place to accommodate all the related file-assets that conform the CAF.

Also fixes a problem that was never evident on a FileAssetContainerUtil.getHostName
that method takes the template-name and extracts the site-name from it. The problem was never evident until I tested against a different site other than demo.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java [hunk 1]
```diff
@@ -416,4 +429,22 @@ private String toString (final FileAsset fileAsset) {
...
+    public boolean isFileAssetContainer(final Contentlet contentlet) {
+        if (null == contentlet || !contentlet.isFileAsset()) {
+            return false;
+        }
+        try {
+            final FileAsset asset = APILocator.getFileAssetAPI().fromContentlet(contentlet);
+            final Folder parentFolder = APILocator.getFolderAPI()
+                    .find(asset.getFolder(), APILocator.systemUser(), false);
+
+            return (CONTAINER_META_INFO.equals(asset.getFileName()) && UtilMethods
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java [hunk 2]
```diff
@@ -106,24 +111,32 @@ public boolean isDataBaseContainerId(final String containerIdentifier) {
...
-        final int startsHost = null != path?     path.indexOf(HOST_INDICATOR): -1;
-        final int endsHost   = -1 != startsHost? path.indexOf(FORWARD_SLASH, startsHost+HOST_INDICATOR.length()): -1;
-        return startsHost != -1 && endsHost != -1?
-                path.substring(startsHost+HOST_INDICATOR.length(), endsHost): null;
+        tmp = tmp.replaceAll(HOST_INDICATOR,"");
+        tmp = tmp.substring(0,tmp.indexOf(CONTAINER_FOLDER_PATH));
+        final String finalString = tmp;
+        Logger.warn(FileAssetContainerUtil.class, ()->  String.format(" extracted hostName `%s`",finalString));
+
+        return (UtilMethods.isSet(tmp) ? tmp : null);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 3]
```diff
@@ -6596,21 +6592,25 @@ public Contentlet copyContentlet(final Contentlet sourceContentlet, final Host h
             }
         }
 
-        for(Contentlet con : versionsToMarkWorking){
+        for(final Contentlet con : versionsToMarkWorking){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: renamte to contentlet or so

**Review by jdotcms — APPROVED** [review]

**Comment by jgambarios** [comment]
@fabrizzio-dotCMS we have failing tests on this PR ☝️

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java [hunk 4]
```diff
@@ -106,24 +111,42 @@ public boolean isDataBaseContainerId(final String containerIdentifier) {
...
-        final int endsHost   = -1 != startsHost? path.indexOf(FORWARD_SLASH, startsHost+HOST_INDICATOR.length()): -1;
-        return startsHost != -1 && endsHost != -1?
-                path.substring(startsHost+HOST_INDICATOR.length(), endsHost): null;
+            tmp = tmp.replaceAll(HOST_INDICATOR, "");
+            tmp = tmp.substring(0, tmp.indexOf(CONTAINER_FOLDER_PATH));
+            final String finalString = tmp;
+            Logger.warn(FileAssetContainerUtil.class,
+                    () -> String.format(" extracted hostName `%s`", finalString));
+
+            return (UtilMethods.isSet(tmp) ? tmp : null);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java [hunk 5]
```diff
@@ -416,4 +439,22 @@ private String toString (final FileAsset fileAsset) {
...
+    public boolean isFileAssetContainer(final Contentlet contentlet) {
+        if (null == contentlet || !contentlet.isFileAsset()) {
+            return false;
+        }
+        try {
+            final FileAsset asset = APILocator.getFileAssetAPI().fromContentlet(contentlet);
+            final Folder parentFolder = APILocator.getFolderAPI()
+                    .find(asset.getFolder(), APILocator.systemUser(), false);
+
+            return (CONTAINER_META_INFO.equals(asset.getFileName()) && UtilMethods
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)

**Pull request merged**